client/driver/spawn/spawn.go

```diff
@@ -285,3 +285,21 @@ func (s *Spawner) readExitCode() (int, error) {
 
 	return exitStatus.ExitCode, nil
 }
+
+// Valid checks that the state of the Spawner is valid and that a subsequent
+// Wait could be called. This is useful to call when reopening a Spawner
+// throught client restarts. If Valid a nil error is returned.
+func (s *Spawner) Valid() error {
+	// If the spawner is still alive, then the task is running and we can wait
+	// on it.
+	if s.Alive() {
+		return nil
+	}
+
+	// The task isn't alive so check that there is a valid exit code file.
+	if _, err := s.readExitCode(); err == nil {
```

Comment:
The reason I didn't was I wanted to have a friendly error code in case it gets exposed to the user. Thoughts?